### DISCUSSION TITLE
Edit Homepage?

### DISCUSSION THREAD

**Discussion by satguys** [mcguirepr89/BirdNET-Pi#1087] [Q&A]
Just installed this tonight, can't wait to wake up tomorrow and see what birds it captures. :)

Anyways I was wondering if there is a way to update the Homepage,  It currently shows my hostname of "birdnetpi1". can I change that to my public URL?  Or better yet y public URL and the location of my Birdnet station, and maybe an explanation of what the site is?

Is there a guide anywhere for editing this stuff?

After a week or two of collecting data I plan to put this on my weather website for my visitors to access.

Thanks!

**Comment by pddpauw** [comment]
yes, you can in fact edit many things.
Have a look at index.php in homepage/ directory.

Not a 100% sure what you want to change, but either you change site name in birdnet.conf file,
or you change directly in index.php.

Just try something an reload...  as a tip (in case less experienced): make a copy of the file before changing things.




`

**Comment by Svardsten53** [comment]
It is quite possible to make a lot of changes on different pages of the installation. The easiest way is to enter minor changes manually, but they usually disappear when the program is updated. If you want to make more advanced modifications, it is advantageously done through scripting.
I have done some translation and modification on my setup: [https://birdnet.svardsten.se/](url)

**Comment by stevedee** [comment]
The BIRDNETPI_URL is in the "thisrun.txt" file.
It should be easy to modify "index.php" to display this.

**Reply by stevedee** [reply]
I've just been tinkering with my portable system and it now looks like this:-
![BirdNetPi_portable](https://github.com/mcguirepr89/BirdNET-Pi/assets/1870807/3a6755a3-5aec-406d-a74e-53c1c1ec8784)

BirdNET-Pi builds/updates the data in **_thisrun.txt_** from **_birdnet.conf_**, so you can edit this conf file to add key-value pairs as required.

I added: MY_TEXT="(Lat/Long:"

In the **_index.php_** file I added 3 lines to retrieve MY_TEXT, LATITUDE & LONGITUDE:-

  if($config["SITE_NAME"] == "") {
    $site_name = "BirdNET-Pi";
  } else {
    $site_name = $config['SITE_NAME'];
    //+++++++++=
    $my_text = $config['MY_TEXT'];
    $my_lat = $config['LATITUDE'];
    $my_long = $config['LONGITUDE'];
  }

...then I modified a line to change the page heading:-

 `</div><h2>BirdNET-Pi $site_name $my_text $my_lat / $my_long)</h2>";`


I edit files in a running BirdNET-Pi system using the procedure I've documented here: http://captainbodgit.blogspot.com/2023/12/birdnet-pi-direct-file-editing.html

I hope this is of some use.

**Reply by trophyguy** [reply]
Like someone else said, when the program get updated, whatever you have on your index.php file will be overwritten.
Keep a copy of the file when you change it just in case that happens.

**Reply by stevedee** [reply]
Yes, absolutely right.
My method is to save the modified files, maybe make a simple note of changes, and then use "Meld" on Linux when the time comes to compare my modified files with the updated files. Changes can be merged as long as they don't conflict in some way.

I started working on code to lookup lat/lon co-ords from entered location (place, popular name or post code):-

`#=============================
#GetLatLong.py
#Return location place name, latitude & longitude from place name or post code
#=============================

import os

from geopy.geocoders import Nominatim
from pprint import pprint


def Location(siteLocation):
	# instantiate a new Nominatim client
	app = Nominatim(user_agent="birdnetpi-stevedee")
	#co-ords need to be to 4 decimal places
	locationX = app.geocode(siteLocation).raw
	place = locationX['name']
	lat = str(round(float(locationX['lat']),4))
	lon = str(round(float(locationX['lon']),4))
	return place,lat,lon



#test routine ********************************	
place,lat,lon = Location("Pagham harbour UK")
pprint (place)
pprint (lat)
pprint (lon)
# ********************************************

`
Christmas prep has stopped progress on integrating this, but hope to get back to this by the end of the month!

**Comment by stevedee** [comment]
I needed to modify my Portable BirdNet-Pi so that **_Daily Charts_** would include my location (..._on the day that it was used_).

So I added a routine to the daily_plot.py file to return data from any key-value pair in the birdnet.conf file;

![daily_plot mods](https://github.com/mcguirepr89/BirdNET-Pi/assets/1870807/1796b493-cb09-41a6-bf33-d0989e0c2e1f)


In my case I'm using this routine to grab the location and the current _confidence_ threshold. Its then just a case of adding these variables into the line that displays the chart title.

This mod changes the Overview display, and as the Daily Charts are basically a snapshot of the Overview, the Daily Charts will also include the location.

A more detailed explanation on my post here: http://captainbodgit.blogspot.com/2024/01/birdnet-pi-portable-chart-info-mods.html
